indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -76,10 +78,21 @@
 
   @Inject
   public TaskLockbox(
-      TaskStorage taskStorage
+      TaskStorage taskStorage,
+      ServerConfig serverConfig
+  )
+  {
+    this.taskStorage = taskStorage;
+    this.lockTimeoutMillis = serverConfig.getMaxIdleTime().getMillis();
+  }
+
+  public TaskLockbox(
+      TaskStorage taskStorage,
+      long lockTimeoutMillis
```

Comment:
Sure, I'll create a separate PR where client can send locktimeout, if not default `serverConfig.getMaxIdleTime()` will be used.